### ISSUE TITLE
Can't send mail, weblate doesn't honour /etc/ssl/openssl.cnf

### ISSUE THREAD

**Issue by plambe** [WeblateOrg/docker#1413]
### Describe the issue

Hi,
I've been trying to make weblate work with an old mail server, with a DH key that is too small for newer distros, with stricter crypto requirements. 8 days ago I opened this https://github.com/WeblateOrg/docker-compose/issues/131, but it was in the wrong repo and I had a mistake in the way I was doing things. 
Here's a sample of the error I'm getting when trying to send a letter due to user registration:
```
celery-notify stderr | [2022-02-25 16:39:15,746: WARNING/398] Failed to send notifications: SSLError: [SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:997)
```
Today, I tried patching the docker image, but with no luck. I added the openssl.cnf file in a clone of this repo (https://github.com/WeblateOrg/docker, release 4.10.1-1) under etc/ssl/openssl.cnf. Here's how it looks now:
```
# docker-compose -f docker-compose-https.yml -f docker-compose-https.override.yml up -d ; docker exec -it weblate-docker_weblate_1 cat /etc/ssl/openssl.cnf | tail -3 ; docker logs --tail 5 -f weblate-docker_weblate_1
weblate-docker_cache_1 is up-to-date
weblate-docker_git-server_1 is up-to-date
weblate-docker_https-portal_1 is up-to-date
weblate-docker_database_1 is up-to-date
Recreating weblate-docker_weblate_1 ... done
[system_default_sect]
MinProtocol = SSLv1.0
CipherString = DEFAULT@SECLEVEL=0
Starting...
```
Notice how `docker exec -it weblate-docker_weblate_1 cat /etc/ssl/openssl.cnf | tail -3` is executed only after this docker container is recreated, so before weblate is run. I still get the same error message though:
```
check stderr | ?: (weblate.E003) Cannot send e-mail ([SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:997)), please check EMAIL_* settings.
```
and if I try to register a new user, so causing an attempt at sending a letter, the same happens: 
```
celery-notify stderr | [2022-02-25 16:50:44,559: WARNING/398] Failed to send notifications: SSLError: [SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:997)
celery-notify stderr | [2022-02-25 16:50:44,559: WARNING/ForkPoolWorker-1] Failed to send notifications: SSLError: [SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:997)
```
Here's openssl's output as well, it doesn't complain: 
```
# docker exec -it weblate-docker_weblate_1 bash -c "openssl s_client -connect mail.somewhere.tld:587 -starttls smtp < /dev/null | grep 'dh key'"
depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = R3
verify return:1
depth=0 CN = mail.somewhere.tld
verify return:1
250 SIZE 0
DONE
```
I suppose I might have made a mistake like in https://github.com/WeblateOrg/docker-compose/issues/131, however I don't see it. 
Any input would be appreciated! 
Regards,
Plamen

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. Patch the docker image by cloning this repo and adding etc/ssl/openssl.cnf which allows for insecure DH key
2. Recreate the docker container by using docker-compose and the newly built image
3. Use an ancient mail server in weblate's conf that has a small DH key

### Expected behavior

Once the patched openssl.cnf file is in an image, weblate should stop complaining about the DH key being too small

**Comment by nijel** [comment]
Maybe the section name is the issue? See https://askubuntu.com/questions/1233186/ubuntu-20-04-how-to-set-lower-ssl-security-level/1265095#1265095

**Comment by plambe** [comment]
It's not the section name, I checked that (because I had found the link you gave) already before opening this issue here. Today I redid everything from scratch because I was working with the presumption I did some mistake somewhere due to being tired. It still doesn't work. 

Finally, I found an ugly workaround - I've configured our SMTP server to allow mail relaying by IP and added our weblate instance's IP in the SMTP server's conf. Now sending letters from weblate works because there's absolutely no crypto, it's plain text. I believe this is worse than using a small DH key, but whatever. I've escalated the fact we need to update our SMTP server, or at least its crypto keys, hopefully some day this will be the permanent solution. 

Since my task was to update my employer's weblate and this is done, I'm going to stop working on this topic and am closing this issue. 

@nijel, thank you for the input both here and in my previous issue! It's appreciated!
